space-concordia-robotics/robotics-prototype
# Joy node 545

# Assignee Section

## Description
Create the ability to change the layout of the joystick controller by changing a configuration file without changing the code.

- Create a node that reads from Joy Node and publishes Arm Commands and Rover Commands(Wheel).
- Node reads from a Yaml configuration file that has the mapping of buttons to commands.

### Steps for Testing
1. Run ros core.
2. Connect the controller via USB.
3. Run Joy Node.
4. Run mcu controller node.
5. Echo the Arm topic.
6. Echo the Rover topic.
7. Press buttons on the controller and watch the arm and rover commands being published.
8. Switch between Arm and Wheel with select and share buttons.
[Needs more details]
When testing on the rover arm and wheel should move.

closes #545 
closes #573 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
